app/src/main/java/org/oppia/android/app/topic/revisioncard/RevisionCardViewModel.kt

```diff
@@ -32,6 +34,53 @@ class RevisionCardViewModel @Inject constructor(
     processRevisionCardLiveData()
   }
 
+  private val topicLiveData: LiveData<AsyncResult<EphemeralTopic>> by lazy {
+    getTopicResultLiveData()
+  }
+
+  private fun getTopicResultLiveData(): LiveData<AsyncResult<EphemeralTopic>> {
+    return topicController.getTopic(profileId, topicId).toLiveData()
+  }
+
+  val nextSubtopicLiveData: LiveData<EphemeralSubtopic> by lazy {
+    Transformations.map(topicLiveData, ::processNextSubtopicData)
+  }
+
+  val previousSubtopicLiveData: LiveData<EphemeralSubtopic> by lazy {
+    Transformations.map(topicLiveData, ::processPreviousSubtopicData)
+  }
+
+  private fun processPreviousSubtopicData(
+    topicLiveData: AsyncResult<EphemeralTopic>
+  ): EphemeralSubtopic? {
+    return if (subtopicId == 0) EphemeralSubtopic.getDefaultInstance()
```

Comment:
Is this check actually needed? It seems like the code below will result in a default instance being returned, anyway.

Reply:
I think this may still be unresolved.